packages/nimble-components/src/table/components/cell/template.ts

```diff
@@ -10,10 +10,18 @@ import {
 // prettier-ignore
 export const template = html<TableCell>`
     <template role="cell">
-        <div ${ref('cellContentContainer')} class="cell-content-container"></div>
+        ${when(x => x.cellViewTag, x => {
+        const cellViewTag = x.cellViewTag!;
+        return html<TableCell>`<${cellViewTag}
+                        class="cell-view"
+                        :cellRecord="${y => y.cellState?.cellRecord}"
+                        :columnConfig="${y => y.cellState?.columnConfig}"
+                    >
+                    </${cellViewTag}>`;
+    })}
```

Comment:
Similar concern as expressed here: https://github.com/ni/nimble/pull/1139#discussion_r1152594197

i.e. we should only create an instance of the `html` template exactly once for a tag name string (here it runs per cell instance). As expressed there we should do that generation on the TableColumn where we can guarantee it will only reconstruct exactly once. The constructed template instance is what we pass through the rest of the system and not the tag name.

Reply:
Updated with that change. `cell/template.ts` exports a `createCellViewTemplate` function, which is used by TableColumn when `cellViewTag` changes.

Is there a better way than this to consume that template from another template (`cell/template.ts`)?
```ts
${when(x => x.cellViewTemplate, x => x.cellViewTemplate!)}
```

Reply:
I think direct interpolation should work

```ts
    `<template role="cell">
        ${x => x.cellViewTemplate)}
    </template>`
```

but if not the when usage is reasonable (still doesn't result in constructing a new template object in memory which was my big concern)

Reply:
Direct dynamic interpolation works. Updated.